pkg/releaselock/controller.go

```diff
@@ -200,8 +162,63 @@ func (c *LockReleaseController) syncLockInfo(ctx context.Context, cm *corev1.Con
 	return nil
 }
 
-// verifyNodeExists validates if the given node object has the exact nodeID, and nodeInternalIP.
-func (c *LockReleaseController) verifyNodeExists(node *corev1.Node, expectedGCEInstanceID, expectedNodeInternalIP string) (bool, error) {
+func (c *LockReleaseController) HandleUpdateEvent(ctx context.Context, oldNode *corev1.Node, newNode *corev1.Node) error {
+	start := time.Now()
+	nodeName := newNode.Name
+	cmName := ConfigMapNamePrefix + nodeName
+	cm, err := c.client.CoreV1().ConfigMaps(util.ManagedFilestoreCSINamespace).Get(ctx, cmName, metav1.GetOptions{})
+	duration := time.Since(start)
+	c.RecordKubeAPIMetrics(err, metrics.ConfigMapResourceType, metrics.GetOpType, metrics.ReconcilerOpSource, duration)
+
+	if err != nil {
+		klog.Errorf("Failed to get configmap in namespace %s: %v", util.ManagedFilestoreCSINamespace, err)
+		return err
+	}
+	klog.Infof("Got configmap (%v) in namespace %s", cm, util.ManagedFilestoreCSINamespace)
+
+	data := cm.DeepCopy().Data
+	for key, filestoreIP := range data {
+		_, _, _, _, gceInstanceID, gkeNodeInternalIP, err := ParseConfigMapKey(key)
+		if err != nil {
+			klog.Errorf("Failed to parse configmap key %s: %v", key, err)
+			continue
+		}
+		klog.V(6).Infof("Verifying GKE node %s with nodeId %s nodeInternalIP %s exists or not", nodeName, gceInstanceID, gkeNodeInternalIP)
+		entryMatchesNewNode, err := c.verifyConfigMapEntry(newNode, gceInstanceID, gkeNodeInternalIP)
+		if err != nil {
+			klog.Errorf("Failed to verify GKE node %s with nodeId %s nodeInternalIP %s still exists: %v", nodeName, gceInstanceID, gkeNodeInternalIP, err)
+			continue
+		}
+		entryMatchesOldNode, err := c.verifyConfigMapEntry(oldNode, gceInstanceID, gkeNodeInternalIP)
+		if err != nil {
+			klog.Errorf("Failed to verify GKE node %s with nodeId %s nodeInternalIP %s still exists: %v", nodeName, gceInstanceID, gkeNodeInternalIP, err)
+			continue
+		}
+		klog.Infof("Checked config map entry against old node(matching result %t), and new node(matching result %t)", entryMatchesOldNode, entryMatchesNewNode)
+		if entryMatchesNewNode {
+			klog.V(6).Infof("GKE node %s with nodeId %s nodeInternalIP %s still exists in API server, skip lock info reconciliation", nodeName, gceInstanceID, gkeNodeInternalIP)
+			continue
+		} else if entryMatchesOldNode {
```

Comment:
nit: I think `if` is enough here. [go/tott/651](https://goto.google.com/tott/651)

Reply:
Done. Will upload the fix shortly.